lib/executor/constant_arrival_rate.go

```diff
@@ -200,7 +200,14 @@ func (car *ConstantArrivalRate) Init(ctx context.Context) error {
 
 // Run executes a constant number of iterations per second.
 //
-// TODO: Reuse the variable arrival rate method?
+// TODO: Split this up and make an independent component that can be reused
+// between the constant and ramping arrival rate executors - that way we can
+// keep the complexity in one well-architected part (with short methods and few
+// lambdas :D), while having both config frontends still be present for maximum
+// UX benefits. Basically, keep the progress bars and scheduling (i.e. at what
+// time should iteration X begin) different, but keep everything else the same.
+// This will allow us to implement https://github.com/loadimpact/k6/issues/1386
+// and things like all of the TODOs below in one place only.
```

Comment:
> This comment is a bit too verbose IMO 

Maybe, but consider it an incentive to fix it sooner :sweat_smile: The long comment don't hurt us, especially when we plan to delete them in the next k6 version

> For example, could this be handled by a generic abstraction layer for all executors, which we've discussed in the past?

Probably no :sweat_smile: After all, the different executors _are_ the different possible concrete implementations of the abstraction "schedule work (iterations) on workers (VUs)"... :wink: Though, as the comment verbosely suggests, the `constant-arrival-rate` and the `ramping-arrival-rate` implementation share a lot of implementation details in common, only their UIs (the description and progressbar) and its scheduling (how we calculate at what time to execute each specific iteration) are different. 

> I wouldn't want another vuHandle situation where a couple executors handle execution differently than others.

But all executors handle execution differently than the other, that's the idea :confused: It just so happened that a `vuHandle` abstraction was useful for both the `ramping-vus` and `externally-controlled` executors, due to some of their similarities. It isn't used anywhere else, because it doesn't make sense to use it anywhere else...

Reply:
> After all, the different executors are the different possible concrete implementations of the abstraction "schedule work (iterations) on workers (VUs)"

Hhmm the abstraction is more like "have a Run method and call GetVU/RunOnce/ReturnVU if you like" :smile: While very flexible, it might be too unrestricted, making the implementations more difficult and error-prone. Since we're finding these common patterns among executors, it makes sense to have `vuHandle` and to do the split you suggest in the `TODO`, but maybe there's a more generic interface we could consider. Handling of the progress bars, for example, could probably be simplified and made generic for all executors.

Anyway, we don't have to get into it now. The current comment feels like too much for a `TODO`, but sure, let's leave it if you think it's going to be removed soon anyway. :)

Reply:
It's not just a comment, I also mentioned it in this issue: https://github.com/loadimpact/k6/issues/1386#issuecomment-652903858